src/main/java/net/snowflake/ingest/streaming/internal/LiteralQuoteUtils.java

```diff
@@ -11,6 +18,38 @@
  */
 class LiteralQuoteUtils {
 
+  /** Maximum number of unquoted column names to store in cache */
+  static final int UNQUOTED_COLUMN_NAME_CACHE_MAX_SIZE = 30000;
+
+  /** Cache storing unquoted column names */
+  private static final LoadingCache<String, String> unquotedColumnNamesCache;
+
+  static {
+    unquotedColumnNamesCache =
+        CacheBuilder.newBuilder()
+            .maximumSize(UNQUOTED_COLUMN_NAME_CACHE_MAX_SIZE)
+            .build(
+                new CacheLoader<String, String>() {
+                  @Override
+                  public String load(String key) {
+                    return unquoteColumnNameInternal(key);
+                  }
+                });
+  }
+
+  /**
+   * Unquote column name expected to be used from the outside. It decides is unquoting would be
+   * expensive. If not, it unquotes directly, otherwise it return a value from a loading cache.
+   */
+  static String unquoteColumnName(String columnName) {
+    try {
+      return unquotedColumnNamesCache.get(columnName);
+    } catch (ExecutionException e) {
+      throw new SFException(
+          e, ErrorCode.INTERNAL_ERROR, "Exception thrown while unquoting column name");
```

Comment:
could we add the column name?

Reply:
This line is not needed for caffeine because its `get` method doesn't throw checked exceptions.

Reply:
Fixed